sdk/servicebus/service-bus/test/streamingReceiver.spec.ts

```diff
@@ -1110,3 +1111,296 @@ describe("Streaming", () => {
+    (nonRetryableError as MessagingError).retryable = false;
+    await Promise.all([
+      (receiverClient as any)["_context"].streamingReceiver!.onDetached(nonRetryableError, true),
+      (receiverClient as any)["_context"].streamingReceiver!.onDetached(nonRetryableError, true)
+    ]);
+
+    receivedErrors.length.should.equal(1, "Unexpected number of errors received.");
+  });
+
+  it("doesn't call user's error handler multiple times (retryable)", async function(): Promise<
+    void
+  > {
+    // Create the sender and receiver.
+    await beforeEachTest(TestClientType.UnpartitionedQueue, "receiveAndDelete");
+    // Send a message so we can be sure when the receiver is open and active.
+    await senderClient.send(TestMessage.getSample());
+    const receivedErrors: any[] = [];
+
+    let receiverIsActiveResolver: Function;
+    const receiverIsActive = new Promise((resolve) => {
+      receiverIsActiveResolver = resolve;
+    });
+    // Start the receiver.
+    receiverClient.subscribe({
+      async processMessage() {
+        // Since we've received a message, mark the receiver as active.
+        receiverIsActiveResolver();
+      },
+      async processError(err) {
+        receivedErrors.push(err);
+      }
+    });
+
+    // Wait until we're sure the receiver is open and receiving messages.
+    await receiverIsActive;
+
+    // Simulate onDetached being called multiple times with non-retryable and then retryable errors.
+    const nonRetryableError = translate(new Error(`I break systems.`));
+    (nonRetryableError as MessagingError).retryable = false;
+    const retryableError = new Error("I temporarily break systems.");
+    (retryableError as any).retryable = true;
+    await Promise.all([
+      (receiverClient as any)["_context"].streamingReceiver!.onDetached(nonRetryableError, true),
+      (receiverClient as any)["_context"].streamingReceiver!.onDetached(retryableError)
+    ]);
+
+    receivedErrors.length.should.equal(1, "Unexpected number of errors received.");
+  });
+});
+
+describe("Streaming - disconnects", function(): void {
+  let serviceBusClient: ServiceBusClientForTests;
+  let senderClient: Sender;
+  let receiverClient: Receiver<ReceivedMessageWithLock> | Receiver<ReceivedMessage>;
+
+  before(() => {
+    serviceBusClient = createServiceBusClientForTests();
+  });
+
+  after(async () => {
+    await serviceBusClient.test.after();
+  });
+
+  async function beforeEachTest(testClientType: TestClientType): Promise<void> {
+    const entityNames = await serviceBusClient.test.createTestEntities(testClientType);
+    receiverClient = serviceBusClient.test.getPeekLockReceiver(entityNames);
+    senderClient = serviceBusClient.test.addToCleanup(
+      serviceBusClient.createSender(entityNames.queue ?? entityNames.topic!)
+    );
+  }
+
+  afterEach(async () => {
+    await serviceBusClient.test.afterEach();
+  });
+
+  it("can receive and settle messages after a disconnect", async function(): Promise<void> {
```

Comment:
This test times out. Not able to receive the second message. 
Worked fine in track 1. Yet to find the root cause.

Reply:
### Findings

Upon simulating the disconnect, the error thrown is being perceived as non-retryable by the core-amqp. Which is why the receiver never came back up for receiving the second message.
![image](https://user-images.githubusercontent.com/10452642/80565480-6aa48f00-89a5-11ea-9ec1-273fc76bd592.png)

This is not the case with amqp-common, this error is treated as a generic-error and is wrapped as a MessagingError(and the default retryable flag is set to true).
![image](https://user-images.githubusercontent.com/10452642/80565564-a50e2c00-89a5-11ea-8f28-ec9e12e48176.png)

This is only the difference b/w track 1 and track 2 and is the cause behind the test failure.